arq/utils.py

```diff
@@ -26,7 +30,23 @@ def to_unix_ms(dt: datetime) -> int:
 
 
 def ms_to_datetime(unix_ms: int) -> datetime:
-    return datetime.fromtimestamp(unix_ms / 1000, tz=timezone.utc)
+    """
+    convert milliseconds to datetime, use the timezone in os.environ
+    """
+    tz = None
+    for timezone_key in timezone_keys:
```

Comment:
this would have a massive impact on performance as we would be checking multiple env keys every time we called this function.

Please move the logic of getting the tz out into a new function, then use `functools.cache` to avoid calling this logic multiple times.